src/position-manager/AllowancePositionManager.sol

```diff
@@ -0,0 +1,254 @@
+    if (_creditDelegations[owner][msg.sender][reserveId] == 0) {
+      return;
+    }
+    _updateCreditDelegation({
+      owner: owner,
+      spender: msg.sender,
+      reserveId: reserveId,
+      newCreditDelegation: 0
+    });
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function withdrawOnBehalfOf(
+    uint256 reserveId,
+    uint256 amount,
+    address onBehalfOf
+  ) external returns (uint256, uint256) {
+    _spendWithdrawAllowance({
+      owner: onBehalfOf,
+      spender: msg.sender,
+      reserveId: reserveId,
+      amount: amount
+    });
+
+    IERC20 asset = _getReserveUnderlying(reserveId);
+    (uint256 withdrawnShares, uint256 withdrawnAmount) = ISpokeBase(SPOKE).withdraw(
+      reserveId,
+      amount,
+      onBehalfOf
+    );
+    asset.safeTransfer(msg.sender, withdrawnAmount);
+
+    return (withdrawnShares, withdrawnAmount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function borrowOnBehalfOf(
+    uint256 reserveId,
+    uint256 amount,
+    address onBehalfOf
+  ) external returns (uint256, uint256) {
+    _spendCreditDelegation({
+      owner: onBehalfOf,
+      spender: msg.sender,
+      reserveId: reserveId,
+      amount: amount
+    });
+
+    IERC20 asset = _getReserveUnderlying(reserveId);
+    (uint256 borrowedShares, uint256 borrowedAmount) = ISpokeBase(SPOKE).borrow(
+      reserveId,
+      amount,
+      onBehalfOf
+    );
+    asset.safeTransfer(msg.sender, borrowedAmount);
+
+    return (borrowedShares, borrowedAmount);
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function withdrawAllowance(
+    address owner,
+    address spender,
+    uint256 reserveId
+  ) external view returns (uint256) {
+    return _withdrawAllowances[owner][spender][reserveId];
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function creditDelegation(
+    address owner,
+    address spender,
+    uint256 reserveId
+  ) external view returns (uint256) {
+    return _creditDelegations[owner][spender][reserveId];
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function DOMAIN_SEPARATOR() external view returns (bytes32) {
+    return _domainSeparator();
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function WITHDRAW_PERMIT_TYPEHASH() external pure returns (bytes32) {
+    return EIP712Hash.WITHDRAW_PERMIT_TYPEHASH;
+  }
+
+  /// @inheritdoc IAllowancePositionManager
+  function CREDIT_DELEGATION_TYPEHASH() external pure returns (bytes32) {
+    return EIP712Hash.CREDIT_DELEGATION_TYPEHASH;
+  }
+
+  function _spendWithdrawAllowance(
+    address owner,
+    address spender,
+    uint256 reserveId,
+    uint256 amount
+  ) internal {
+    uint256 currentAllowance = _withdrawAllowances[owner][spender][reserveId];
+    require(currentAllowance >= amount, InsufficientWithdrawAllowance(currentAllowance, amount));
+    _withdrawAllowances[owner][spender][reserveId] = currentAllowance.uncheckedSub(amount);
```

Comment:
should we sentinel values here? 
uint.max == infinite allowance, never decreased, like erc20 implementations 
uint.max - 1 == one time infinite allowance

Reply:
We could consider this logic yeah, that would solve your question https://github.com/aave/aave-v4/pull/1054#discussion_r2632482450 too @miguelmtzinf